src/main/java/org/wildfly/security/auth/realm/ldap/UserPasswordPasswordUtil.java

```diff
@@ -285,4 +285,20 @@ private static void composeBsdCryptBasedPassword(ByteArrayOutputStream out, BSDU
 
         out.write(ByteIterator.ofBytes(password.getHash()).base64Encode(Base64Alphabet.MOD_CRYPT, false).asUtf8().drain());
     }
+
+    public static boolean isAlgorithmSupported(String algorithm) {
+        return ALGORITHM_SIMPLE_DIGEST_MD5.equals(algorithm) ||
```

Comment:
This is a lot of ``equals()`` calls.  I think a String switch would probably be better here.